### PR TITLE
PyPi Support

### PR DESCRIPTION
**Pull request by AnonGuy**
You'll need the variables `PyPi.Username` `PyPi.Password` as credentials for https://pypi.org.

### CODE REVIEW

**Comment by AnonGuy** [comment]
It might also be worth adding a build trigger for when a [particular tag](https://git-scm.com/book/en/v2/Git-Basics-Tagging) is released, then running `poetry version $(tagName)`. Poetry won't publish a new version if the semver hasn't changed.

**Comment by thebeanogamer** [comment]
👍  This is great work, I'll set the environment variables, but I do have a couple of questions:

1. Do I need to manually create the project on PyPi, or will this do that for me?
2. My current workflow involves me squashing in GitHub, then pulling `master`, tagging it, then pushing it back, I assume there is a nicer way to do this?
3. The build pipeline is [here](https://github.com/thebeanogamer/hstsparser/blob/master/azure-pipelines/release-pipeline.yml), I'm assuming that `poetry version $(tagName)` wont actually work  so it would have to be something more like `poetry version $(git tag -l --points-at HEAD)`?

File: README.md [hunk 1]
```diff
@@ -6,30 +6,31 @@ HSTS Parser is a simple tool to parse Firefox and Chrome's HSTS databases into a
...
 (Invoke-WebRequest -Uri https://raw.githubusercontent.com/python-poetry/poetry/master/get-poetry.py -UseBasicParsing).Content | python
-poetry install
+poetry install --no-dev
 ```
 
 Alternatively, if you're using Windows, you can use the executables on the [releases page](https://github.com/thebeanogamer/hstsparser/releases/latest) to not need to install anything at all.
 
 ## Usage
 
-All of the below documentation is written for the Python version rather than the standalone, but the commands are the same, just replacing `poetry run python hstsparser.py` with the name of the executable.
```
Comment (thebeanogamer) [inline README.md]: I'm a bit iffy on removing this line, but if it did stay it would obviously need to be expanded.

**Comment by thebeanogamer** [comment]
Environment variables set

**Comment by thebeanogamer** [comment]
Also:

```shell
$ poetry run hstsparser

[ValueError]
not enough values to unpack (expected 2, got 1)
```

Is this because it isn't a module?

**Comment by AnonGuy** [comment]
> Do I need to manually create the project on PyPi, or will this do that for me?

It'll create the project automatically, if you've provided valid credentials.
> My current workflow involves me squashing in GitHub, then pulling `master`, tagging it, then pushing it back, I assume there is a nicer way to do this?

I've not really done much work with tagged releases, but I'd assume you can tag commits before pushing them to github?
> I'm assuming that `poetry version $(tagName)` wont actually work so it would have to be something more like `poetry version $(git tag -l --points-at HEAD)`?

I assumed azure would expose some variable for the current tag name, but if not I'm sure you could use something like that. Just as long as the tag name is [semver compliant](https://semver.org/).

**Comment by thebeanogamer** [comment]
> I've not really done much work with tagged releases, but I'd assume you can tag commits before pushing them to github?
Yeah, but if `master` is a protected branch, AFAIK I can only edit it from GitHub.com, and you can't create tags in the web interface.
> I assumed azure would expose some variable for the current tag name, but if not I'm sure you could use something like that. Just as long as the tag name is semver compliant.
Not actually sure... Will look into this

**Comment by AnonGuy** [comment]
> Also:
> 
> ```shell
> $ poetry run hstsparser
> 
> [ValueError]
> not enough values to unpack (expected 2, got 1)
> ```
> 
> Is this because it isn't a module?

I'm not sure why this happens, I think it might be caused by me not using the script entry points correctly. For now you should be able to just use the `hstsparser` command within `poetry shell`.

**Comment by AnonGuy** [comment]
I'll see if I can fix the entry point issue later tonight.

**Comment by thebeanogamer** [comment]
> I'll see if I can fix the entry point issue later tonight.

🙌

**Comment by thebeanogamer** [comment]
Is it worth making PyPi and GitHub Releases into seperate jobs so that the pipeline is faster and isn't including dev dependencies in the exe?

**Comment by thebeanogamer** [comment]
It's kinda hacky, but adding the below function and settings the script to `hstsparser:module` seems to work. I'd assume there is a better way to do it though...

```python
def module():
    pass
```

**Comment by thebeanogamer** [comment]
Huh, apparently tags don't respect protected branches...

**Comment by thebeanogamer** [comment]
Only thing left to do is see if there is a nicer way to do the `script`, then this is ready to merge I think.

File: hstsparser.py [hunk 2]
```diff
@@ -123,78 +123,90 @@ def date_round(date: datetime.datetime) -> datetime.datetime:
...
-                "Expiry",
-                "Include Subdomains",
-                "Last Observed",
-                "Cracked Hash",
-            ],
-        )
-    else:
+def main() -> None:
+    """
+    Entry point for command line alias.
```
Comment (AnonGuy) [inline hstsparser.py]: Not sure if it applies to this project, but [Flake8 D200](http://www.pydocstyle.org/en/5.0.2/error_codes.html) states that a one-line docstring should fit on one line...

Comment (thebeanogamer) [reply hstsparser.py]: So it does. I'll move them all to one line then, surprised Flake8 didn't catch that...

**Pull request merged**